test/type_check/internals/parser_test.exs

```diff
@@ -0,0 +1,412 @@
+defmodule TypeCheck.Internals.ParserTest do
+  @moduledoc false
+  use ExUnit.Case, async: true
+  doctest TypeCheck.Internals.Parser
+
+  alias TypeCheck.Internals.Parser
+  alias TypeCheck.Builtin, as: B
+
+  alias TypeCheck.Internals.ParserTest.TypespecSample
+
+  # test that fetch_spec doesn't explode and all specs are supported
+  describe "fetch_spec smoke" do
```

Comment:
I love how much testing you're able to do with this little code!
Hooray for reflection :blush: